robocop/checkers/naming.py

```diff
@@ -653,7 +671,9 @@ def visit_VariableSection(self, node):  # noqa
             if not child.data_tokens:
                 continue
             token = child.data_tokens[0]
-            if token.type == Token.VARIABLE and token.value and not token.value.isupper():
+            var_name = search_variable(token.value).base
+            normalized_var_name = remove_nested_variables(var_name)
+            if token.type == Token.VARIABLE and token.value and not normalized_var_name.isupper():
```

Comment:
Do we need to assign var_name at all if token.type != Token.VARIABLE?

Reply:
You're right, I fixed that. Is it better now?

Reply:
Wait a sec, I broke something.

Reply:
Well, I found a bug in Robot Framework, so this PR needs to wait.